source/components/defining-a-component.md

```diff
@@ -81,3 +81,63 @@ example, if you have a component called `blog-post`, you would create a
 file at `app/components/blog-post.js`. If your component was called
 `audio-player-controls`, the file name would be at
 `app/components/audio-player-controls.js`.
+
+### Dynamically rendering a component
+
+The `{{component}}` helper can be used to defer the selection of a component to
+run time. The `{{my-component}}` syntax would always render the same component,
+whereas using the `{{component}}` helper allows swapping the component rendered
+on the fly. This is useful in cases where, for example, you want to interact
+with different external libraries depending on the data. Using the `{{component}}`
+helper would allow you to keep those different logic well-separated.
+
+The first parameter of the helper is the name of a component to render, as a string. So if you have `{{component 'blog-post'}}`, that is just the same as just `{{blog-post}}`.
+
+The real value of `{{component}}` comes from being able to dynamically pick
+the component being rendered. Below is an example of using the helper as a
+mean to dispatch to different components for displaying different kinds of posts:
+
+
+```app/templates/components/foo-component.hbs
+<h3>Hello from foo!</h3>
+<p>{{post.body}}</p>
+```
+
+```app/templates/components/bar-component.hbs
+<h3>Hello from bar!</h3>
+<div>{{post.author}}</div>
+```
+
+```app/routes/index.js
+var posts = [{
+    componentName: 'foo-component',  // key used to determine the rendered component
+    body: "There are lots of à la carte software environments in this world."
+  }, {
+    componentName: 'bar-component',
+    author: "Drew Crawford"
+}];
+
+export default Ember.Route.extend({
+  model: function() {
```

Comment:
I know about that syntax, and use it at work everyday. I picked the old syntax since code from the previous section used it. Anyways, updated :)